### ISSUE TITLE
drive: Service account with wrong credentials doesn't return errror when listing root

### ISSUE THREAD

**Issue by zenjabba** [rclone/rclone#3763]
```
rclone size hbg: --drive-service-account-file=/opt/sa/1.json
Total objects: 0
Total size: 0 Bytes (0 Bytes)

```
```
rclone size hbg:
Total objects: 20001
Total size: 20.623 TBytes (22675191555624 Bytes)
```
```
rclone lsd hbg: --drive-service-account-file=/opt/sa/1.json
echo $?
0
```
```
rclone lsd hbg:
-1 2019-11-07 00:12:05        -1 hbg
echo $?
0
```

**Comment by darthShadow** [comment]
Just to clarify, this issue is about the exit code. I believe a non-zero exit code needs to be returned when the auth with the specific service account fails.

**Comment by 88lex** [comment]
Further to the above:

This can be extraordinarily dangerous.
****************
`rclone size/lsd/etc` does not show an error. Which is inconvenient.

But if you run `rclone sync` with a service file that exists but does not have sharing/permission on the source Shared Drive then it deletes all the files in the destination.

`rclone sync source: destination:`  <= works perfectly.

`rclone sync source: destination: --drive-service-account-file=1.json` <= works perfectly when 1.json exists and source and destination both have shared files with the email for 1.json.

`rclone sync source: destination: --drive-service-account-file=2.json` <= DELETES all the files in destination when 2.json exists and has permissions in destination but does not have permissions in source.

Tested with current rclone win/linux/mac and observed the same behavior.

**Comment by ncw** [comment]
See discussion in forum: https://forum.rclone.org/t/rclone-missing-error-code-when-sas-have-no-permission/13086

The conclusion was that this is a bug/oddity in the google drive API which causes the API to return empty rather than a permissions error.

To work around this 

when listing the root
- if nothing is returned
- list again with a limit of 1 file and without the parents and report error if any

As listings without setting parents do return a permission denied error.

**Comment by ncw** [comment]
I've attempted to fix this here

https://beta.rclone.org/branch/v1.50.2-071-g11b5ba92-fix-sharedrive-beta/ (uploaded in 15-30 mins)

    This change:
    - calls teamdrives get on rclone about
    - calls teamdrives get on a listing of the root which returned no entries
    
    These will both detect a team drive which has the incorrect auth and
    workaround the issue.

**Comment by 88lex** [comment]
@ncw Thank you. Tested and working on Win10 and Ubuntu.

Blank response with exit code 0 shown for team/shared drive remote that has id/secret auth. 
Error message and 404 for same remote using a service account that is not authorized.

```
rclone about remote:

rclone about remote: --drive-service-account-file=/sa/1.json
2019/11/28 10:00:46 Failed to about: About call failed: failed to get Team/Shared Drive info: googleapi: Error 404: Shared drive not found: 0123456789VA, notFound
```

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.51

Thanks for testing :-)

**Comment by 88lex** [comment]
I remain astonished how quickly you and your gang fix so many bugs and add so many features... for an open source project. Especially given the number of back end platforms you support and the amazing depth of functionality. Ain't easy.

Pulled the latest beta on Mac/Ubuntu/Win and is working as expected, thank you. Saved me a load of workaround code in my scripts :-)
